File: Java/src/com/leetcode/Easy/SetMismatch.java
package com.leetcode.Easy;

import java.util.*;

// 645. Set Mismatch
public class SetMismatch {
    public int[] findErrorNums(int[] nums) {
        int sum = 0, dup = -1;
        List<Integer> list = new ArrayList<>();
        for (int num : nums) {
            if (list.contains(num) && dup == -1) {
                dup = num;
            } else {
                list.add(num);
                sum += num;
            }
        }
        if (dup == -1) {
            return new int[]{};
        } else {
            int n = list.size();
            int expSum = (n + 1) * (n + 2) / 2;
            int miss = expSum - sum;
            return new int[]{dup, miss};
        }
    }
}

File: JavaScript/Easy/SetMismatch.js
/**
 * 645. Set Mismatch
 * @param {number[]} nums
 * @return {number[]}
 */
const findErrorNums = function (nums) {
  let set = new Set();
  let len = nums.length, duplicate = 0;
  let sum = (len * (len + 1)) / 2;
  for (let num of nums) {
    if (set.has(num))
      duplicate = num;

    sum -= num;
    set.add(num);
  }

  return [duplicate, sum + duplicate];
};